src/Features/Core/Portable/MetadataAsSource/TelemetryMessage.cs

```diff
@@ -32,10 +36,37 @@ public void SetSourceFileSource(string source)
             _sourceFileSource = source;
         }
 
+        public void SetReferenceAssembly(string referenceAssembly)
+        {
+            _referenceAssembly = referenceAssembly;
+        }
+
+        public void SetDll(string dll)
+        {
+            _dll = dll;
+        }
+
+        public void SetDecompiled(bool decompiled)
+        {
+            _decompiled = decompiled;
+        }
```

Comment:
Pretty much Java-stylish 😃 .

I see you're consistent with the already existing fields, but curious if there was a reason these are not properties?

Reply:
Nobody reads them, and write-only properties are weird 🤷‍♂️

Reply:
@davidwengier Aren't they read in `SetLogProperties` below?

Reply:
The fields are read, the properties wouldn't be. But I really don't care, I can change it if you like

Reply:
> The fields are read, the properties wouldn't be. But I really don't care, I can change it if you like

Ahhh because of nullable? I missed that. No, definitely the current is better than a write only property (which violates [CA1044](https://learn.microsoft.com/en-us/dotnet/fundamentals/code-analysis/quality-rules/ca1044)). I was just curious to know the reason since and I see now!


Thanks @davidwengier